integration/app/test-demo/src/pages/VideoFilterTestApp.tsx

```diff
@@ -0,0 +1,109 @@
+    blurStrength: 20,
+    filterCPUUtilization: 20,
+    reportingPeriodMillis: 1000,
+  };
+  const [backgroundBlurOptions, setBackgroundBlurOptions] = useState<BackgroundBlurOptions>(defaultOptions);
+
+  const setBlurStrength = (newBlurStrength: number): void => {
+    console.log(`Setting blur strength to ${newBlurStrength} - should re-initialize the bg blur provider`);
+    const options = {...backgroundBlurOptions, blurStrength : newBlurStrength};
+    setBackgroundBlurOptions(options);
+  }
+
+  return(
+    <div style={{ display: 'flex', flexDirection: 'column', width: '70%', margin: 'auto' }}>
+      <BackgroundBlurProvider options={backgroundBlurOptions}>
+        <MeetingProvider {...{logLevel: LogLevel.INFO}}>
+        <h3 data-testid='app-name'>Video Filter Test</h3>
+        <MeetingForm />
+        <Meeting onBlurStrengthChanged={(newBlurStrength) => setBlurStrength(newBlurStrength)}/>
+        </MeetingProvider>
+      </BackgroundBlurProvider>
+    </div>
+    )
+};
+
+interface Props {
+  onBlurStrengthChanged: (blurStrength: number) => void;
+}
+
+const Meeting: React.FC<Props> = ({ onBlurStrengthChanged }) => {
+  const { isBackgroundBlurSupported, createBackgroundBlurDevice } = useBackgroundBlur();
+  const meetingManager = useMeetingManager();
+  const { selectedDevice } = useVideoInputs();
+  const { toggleVideo } = useLocalVideo();
+  const meetingStatus = useMeetingStatus();
+  const [ blurStrength, setBlurStrength ] = useState<string>("20");
+
+  useEffect(() => {
+    const addBackgroundBlur = async () => {
+      if (
+        isBackgroundBlurSupported &&
+        meetingStatus === MeetingStatus.Succeeded
+      ) {
+        console.log("Creating background blur device called");
+        const chosenVideoTransformDevice = await createBackgroundBlurDevice(selectedDevice);
+        console.log(chosenVideoTransformDevice);
+        await meetingManager.selectVideoInputDevice(chosenVideoTransformDevice);
+        toggleVideo();
+      }
+    };
+    addBackgroundBlur();
+  }, [
+    isBackgroundBlurSupported,
+    meetingStatus,
+  ]);
+
+  const updateBlurStrength = (e: ChangeEvent<HTMLSelectElement>) => {
+    e.preventDefault();
+    setBlurStrength(e.target.value);
+    console.log('Updated to blurStrength:' + e.target.value);
+    const newBlurStrength = parseInt(e.target.value);
+    onBlurStrengthChanged(newBlurStrength);
+  }
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', width: '70%', margin: 'auto' }}>
+    <MeetingInfo />
+    <div style={{height: '30rem', width:`100%` }}>
+      <VideoTileGrid />
+    </div>
+    <ControlBar
+      layout='undocked-horizontal'
+      showLabels
+    >
+      <FormField
+        field={Select}
+        label="Blur Strength"
+        value={blurStrength}
+        onChange={updateBlurStrength}
+        options={[{value: '20', label: '20'}, {value: '40', label: '40'}, {value: '60', label: '60'}]}
+      />
```

Comment:
The ControlBarButton doesnt have the click event in the callback, but I need to know which blur strength was selected. So thats why I cant use the control bar button

Reply:
Can we use the `popOver` of `ControlBarButton` to do this? Like how we switching the input devices.

Reply:
Its not really the same use case - A control bar button is a button that also has a dropdown -  I just need the dropdown, which is the FormField. Just because its called ControlBar and ControlBarButton doesnt mean you cant have other items in the control bar

Reply:
Plus, we want to be testing our components here, so I'm using the VideoInputBackgroundBlurControl component to enable and disable BG blur. This component doesnt have blurStrength options, so we jsut need a dropdown to change the blurStrength, but we shouldnt use a separate component to enable background blur - because that means were not testing the VideoInputBackgroundBlurControl component

Reply:
Gotcha, this make sense, thanks for answering my question. 👍